metrics/metrics.go

```diff
@@ -38,6 +38,38 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		}, []string{"result", "strategy", "namespace", "node"})
 
+	PodsEvictedSuccess = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      DeschedulerSubsystem,
+			Name:           "pods_evicted_success",
+			Help:           "Number of pods successfully evicted per strategy",
+			StabilityLevel: metrics.ALPHA,
+		}, []string{"strategy", "namespace"})
+
+	PodsEvictedFailed = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      DeschedulerSubsystem,
+			Name:           "pods_evicted_failed",
+			Help:           "Number of pods failed to evict per strategy",
+			StabilityLevel: metrics.ALPHA,
+		}, []string{"strategy", "namespace"})
+
+	TotalPodsEvicted = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      DeschedulerSubsystem,
+			Name:           "total_pods_evicted",
+			Help:           "Total pods evicted so far",
+			StabilityLevel: metrics.ALPHA,
+		}, []string{"result", "namespace"})
+
+	TotalPodsSkipped = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      DeschedulerSubsystem,
+			Name:           "total_pods_skipped",
+			Help:           "Total pods skipped for a single run",
```

Comment:
I wonder what a single run means in this context? Is it meant as a single [descheduling cycle](https://github.com/kubernetes-sigs/descheduler/blob/901a16ecbcaaf41c11318920f8a047c918b6c30d/pkg/descheduler/descheduler.go#L218)? `TotalPodsSkipped` metric is used only inside `PodEvictor` where there's currently no way to distinguish when a single run finished and a new one started.

Reply:
Maybe we could introduce a copy of these metrics named something like `PodsSkippedLastRun`, for example. It would require some refactoring to the PodEvictor, maybe pass it a timestamp at the start of a new run, and if that timestamp differs from the last one it remembers then reset the counters for those metrics. But I think knowing the totals vs individual runs would be helpful.

I suppose you could get the total by just summing the individual runs, in which case we don't need the `Total..` metrics anymore. Either way, I think this is a good start and the refactors could come as a follow up. Wdyt?

Reply:
With `single run` I meant a single descheduling cycle (somewhat mentioned here in this comment: https://github.com/kubernetes-sigs/descheduler/issues/503#issuecomment-861813502). So just wanted to check will it still be a good idea to introduce this metric or not @damemi @ingvagabund